### PR TITLE
Use label translator to translate groups and tabs

### PR DESCRIPTION
**Pull request by EmmanuelVella**
Currently, when calling ``->with()`` or ``->tab()`` on a form or show mapper, the label **is** the exact name you passed : ``->with('myTest')`` would give the ``myTest`` label. This force us to eventually pass a manual label like ``->with('form.group_my_test')`` to get the good output.

With this PR, the label is created just like in the ``add`` method. So ``->with('myTest')`` would give the ``form.group_my_test`` label.

I add a check to try to keep BC when the name contains dots, but I'm not sure about it.

### CODE REVIEW

File: Resources/views/CRUD/base_edit_form.html.twig [hunk 1]
```diff
@@ -33,7 +33,7 @@
                         <div class="nav-tabs-custom">
                             <ul class="nav nav-tabs" role="tablist">
                                 {% for name, form_tab in admin.formtabs %}
-                                    <li{% if loop.index == 1 %} class="active"{% endif %}><a href="#tab_{{ loop.index }}" data-toggle="tab"><span class="glyphicon glyphicon-exclamation-sign has-errors hide"></span> {{ admin.trans(name, {}, form_tab.translation_domain) }}</a></li>
+                                    <li{% if loop.index == 1 %} class="active"{% endif %}><a href="#tab_{{ loop.index }}" data-toggle="tab"><span class="glyphicon glyphicon-exclamation-sign has-errors hide"></span> {{ admin.trans(form_tab.label, {}, form_tab.translation_domain) }}</a></li>
```
Comment (Soullivaneuh) [inline Resources/views/CRUD/base_edit_form.html.twig]: Changing the name is BC break AFAIK. Why changing it?

File: Resources/views/CRUD/base_edit_form_macro.html.twig [hunk 2]
```diff
@@ -8,7 +8,7 @@
             <div class="{{ form_group.box_class }}">
                 <div class="box-header">
                     <h4 class="box-title">
-                        {{ admin.trans(form_group.name, {}, form_group.translation_domain) }}
+                        {{ admin.trans(form_group.label, {}, form_group.translation_domain) }}
```
Comment (Soullivaneuh) [inline Resources/views/CRUD/base_edit_form_macro.html.twig]: Same as https://github.com/sonata-project/SonataAdminBundle/pull/2870/files#r32898381

File: Resources/views/CRUD/base_show.html.twig [hunk 3]
```diff
@@ -31,7 +31,7 @@ file that was distributed with this source code.
                     <div class="box-header with-border">
                         <h4 class="box-title">
                             {% block show_title %}
-                                {{ admin.trans(name) }}
+                                {{ admin.trans(view_group.label, {}, view_group.translation_domain) }}
```
Comment (Soullivaneuh) [inline Resources/views/CRUD/base_show.html.twig]: Same as https://github.com/sonata-project/SonataAdminBundle/pull/2870/files#r32898381

File: Resources/views/CRUD/base_edit_form.html.twig (hunk 1 above)
Comment (Soullivaneuh) [inline Resources/views/CRUD/base_edit_form.html.twig]: And BTW, where are the related translations?

**Comment by Soullivaneuh** [comment]
I'll try your PR later to understand the goal.

But on first review, this seems to be completely BC break, isn't it?

Can you please rebase your PR?

**Comment by OskarStark** [comment]
@EmmanuelVella can you please give us feedback?

File: Resources/views/CRUD/base_edit_form.html.twig (hunk 1 above)
Comment (EmmanuelVella) [inline Resources/views/CRUD/base_edit_form.html.twig]: This is explained in the PR comment : currently, when doing `->tab('field')`, the label translation id is `field`. This is not uniform with `->add('field')`, which label translation id is `form.label_field`.

**Comment by EmmanuelVella** [comment]
@Soullivaneuh @OskarStark PR rebased !

@Soullivaneuh This will need to update the translation id in the translation files.

File: Tests/Mapper/BaseGroupedMapperTest.php [hunk 4]
```diff
@@ -152,6 +166,32 @@ public function testEndException()
         $this->baseGroupedMapper->end();
     }
 
+    public function labelDataProvider()
+    {
+        return array(
+            array('fooGroup1',      null,           'label_foogroup1'),
```
Comment (OskarStark) [inline Tests/Mapper/BaseGroupedMapperTest.php]: why all these whitespace?

Comment (EmmanuelVella) [inline Tests/Mapper/BaseGroupedMapperTest.php]: I just fixed this !

Comment (OskarStark) [inline Tests/Mapper/BaseGroupedMapperTest.php]: :+1:

**Comment by OskarStark** [comment]
ping @Soullivaneuh

**Comment by Soullivaneuh** [comment]
I would like to have this one BC.

@EmmanuelVella If you can do it, could you please reopen a PR on the 3.x branch?

**Comment by OskarStark** [comment]
ping @EmmanuelVella

**Comment by core23** [comment]
Ping @EmmanuelVella

**Comment by EmmanuelVella** [comment]
Opening a new PR shortly !

**Pull request closed without merging**